arangod/Graph/BaseOptions.cpp

```diff
@@ -605,11 +619,27 @@ void BaseOptions::isQueryKilledCallback() const {
 }
 
 void BaseOptions::setVertexProjections(Projections projections) {
+  if (!getVertexProjections().empty()) {
+    resourceMonitor().decreaseMemoryUsage(getVertexProjections().size() *
+                                          sizeof(aql::Projections::Projection));
+  }
+  arangodb::ResourceUsageScope guard(
+      resourceMonitor(),
+      projections.size() * sizeof(aql::Projections::Projection));
   _vertexProjections = std::move(projections);
+  guard.steal();  // now we are responsible for tracking the memory
 }
 
 void BaseOptions::setEdgeProjections(Projections projections) {
+  if (!getEdgeProjections().empty()) {
+    resourceMonitor().decreaseMemoryUsage(getEdgeProjections().size() *
+                                          sizeof(aql::Projections::Projection));
+  }
+  arangodb::ResourceUsageScope guard(
```

Comment:
same here

Reply:
Added in: 60044d169796e5dd8850ba2454d5f44463b1fc76